src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj

```diff
@@ -44,18 +45,53 @@
     <None Include="buildTransitive\Sentry.Bindings.Cocoa.targets" Pack="true" PackagePath="buildTransitive\Sentry.Bindings.Cocoa.targets" />
   </ItemGroup>
 
-  <!-- Build the Sentry Cocoa SDK -->
-  <Target Name="_BuildSentryCocoaSDK" BeforeTargets="DispatchToInnerBuilds;BeforeBuild" Condition="'$(CI)' != 'true' and $([MSBuild]::IsOSPlatform('OSX'))"
-    Inputs="..\..\.git\modules\modules\sentry-cocoa\HEAD" Outputs="..\..\modules\sentry-cocoa\Carthage\.built-from-sha">
-    <MSBuild Projects="$(MSBuildProjectFile)" Targets="_InnerBuildSentryCocoaSDK" Properties="TargetFramework=once" />
+  <!-- Downloads and sets up the Cocoa SDK: dotnet msbuild /t:setupCocoaSDK src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj -->
+  <Target Name="SetupCocoaSDK"
+          Condition="$([MSBuild]::IsOSPlatform('OSX')) And !Exists('$(SentryCocoaCache)Sentry-Dynamic.xcframework')"
+          BeforeTargets="BeforeBuild">
+
+    <PropertyGroup>
+      <PropertiesContent>$([System.IO.File]::ReadAllText("../../modules/sentry-cocoa.properties"))</PropertiesContent>
+      <CocoaVersion>$([System.Text.RegularExpressions.Regex]::Match($(PropertiesContent), 'version\s*=\s*(\d+\.\d+\.\d+)').Groups[1].Value)</CocoaVersion>
```

Comment:
this regex will match beta releases but extract only the `n.n.n` and not the `-xyz`

Reply:
Ah, nice catch. @bitsandfoxes you'll want to [fix that in the Unity repo](https://github.com/getsentry/sentry-unity/blob/8de84e71c6439d43ece603880a95d4daa15188a5/Directory.Build.targets#L149) as well.

I've gone with dead simple - which is to match any sequence of non-whitespace characters:
https://github.com/getsentry/sentry-dotnet/blob/bbd1f64417731aab4d780afda0902066a0129a11/src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj#L55

Reply:
@bruno-garcia the issue you found in the regex has been addressed. Any other "changes requested" or can we merge?